satpy/readers/sar_c_safe.py

```diff
@@ -704,3 +689,110 @@ def start_time(self):
     def end_time(self):
         """Get the end time."""
         return self._end_time
+
+
+class SAFESARReader(GenericYAMLReader):
+    """A reader for SAFE SAR-C data for Sentinel 1 satellites."""
+
+    def __init__(self, config, filter_parameters=None):
+        """Set up the SAR reader."""
+        super().__init__(config)
+        self.filter_parameters = filter_parameters
+        self.files_by_type = defaultdict(list)
+        self.storage_items = []
+
+    @property
+    def start_time(self):
+        """Get the start time."""
+        return self.storage_items.values()[0].filename_info["start_time"].replace(tzinfo=tz.utc)
+
+    @property
+    def end_time(self):
+        """Get the end time."""
+        return self.storage_items.values()[0].filename_info["end_time"].replace(tzinfo=tz.utc)
+
+    def load(self, dataset_keys, **kwargs):
+        """Load some data."""
+        if kwargs:
+            raise NotImplementedError(f"Don't know how to handle kwargs {kwargs}")
+        datasets = DatasetDict()
+        for key in dataset_keys:
+            for handler in self.storage_items.values():
+                val = handler.get_dataset(key, info=dict())
+                if val is not None:
+                    val.attrs["start_time"] = handler.start_time
+                    # val.attrs["footprint"] = self.footprint
+                    if key["name"] not in ["longitude", "latitude"]:
+                        lonlats = self.load([DataID(self._id_keys, name="longitude", polarization=key["polarization"]),
+                                             DataID(self._id_keys, name="latitude", polarization=key["polarization"])])
+                        gcps = val.coords["spatial_ref"].attrs["gcps"]
+                        from pyresample.future.geometry import SwathDefinition
+                        val.attrs["area"] = SwathDefinition(lonlats["longitude"], lonlats["latitude"],
+                                                            attrs=dict(gcps=gcps))
+                    datasets[key] = val
```

Comment:
How were swath definitions handled before this PR? Is there no way to do this from the base reader class?

Reply:
Before, the swath definition was created by the base reader as we provided longitude and latitude coordinates. The problem with that is that I want to fine tune it, so as to pass the gcps as a metadata attribute to the swath definition itself, and I didn't see how to do this cleanly with satpy at the moment.

Reply:
Ok. I guess we'll just have to keep this reader in mind when someone comes along with another reader leveraging GCPs.